app/views/layouts/application.html.erb

```diff
@@ -37,10 +37,10 @@
 
       </ul>
       <ul class="nav navbar-nav navbar-right">
-        <% if current_user && current_user.charity_admin? %>
-          <li class="custom-nav"> <%= link_to "Logged in as #{current_user.username}", admin_dashboard_path %> </li>
+        <% if current_user && (current_user.charity_original_admin? || current_user.charity_admin?) %>
```

Comment:
This looks like a very messy conditional to have in the view. Let's make sure we refactor this. We can push this logic down into the model.